### ISSUE TITLE
Message attachments (multiple photos, files, other messages for forwarding, etc)

### ISSUE THREAD

**Issue by MurzNN** [matrix-org/matrix-spec#541]

**Comment by gabrc52** [comment]
Any update in 2023? Most messaging apps work like this, which makes it more convoluted to write bridges to send one message event with the text and one message event for every attachment.

**Comment by turt2live** [comment]
MSC1767 is the start of a very long series of MSCs, and is expected to make further progress as we continue our IETF mission in the MIMI working group - TWIM will contain updates as we progress here (and generally the matrix.org blog).

**Comment by gabrc52** [comment]
> I think being able to send an image with a message attached is an absolute requirement nowadays. So it would be really great to have that feature. Every other messenger has it as well.

Yeah, it is. Thankfully now [MSC2530](https://github.com/matrix-org/matrix-spec-proposals/pull/2530) was approved, so it might make it to clients soon enough.

> But especially on mobile you cannot QUICKLY send a message after the image, because typing on mobile devices is slow.

Could you file an issue for your mobile client if it hasn't been filed?

**Comment by ThiefMaster** [comment]
My second point was more about the fact that on a desktop you could already send the image and quickly write a message, while on a mobile phone it takes longer simply due to the inferior keyboard (onscreen vs full-size physical). So not really a problem on the Matrix/Element/whateverclient side

**Comment by turt2live** [comment]
The spec process requires implementation of an MSC (proposal) before it can reasonably continue through the process. Currently there are many proposals, including ones for Extensible Events, but little to no implementation. We'd really love to see this feature landed and have spent quite a bit of time working on Extensible Events already - if folks are interested in this feature, implementation would be greatly appreciated.

**Comment by NAM1025** [comment]
Very interested in this feature. It's the main thing keeping me from trying to get friends and family to use Matrix/Element.

**Comment by Renbo2024** [comment]
One possible halfway option which would not require protocol changes would be to allow multiple images to be selected by the client, but send them as individual separate messages?

**Comment by VityaSchel** [comment]
this issue turns 7 years this year!! still no galleries in matrix! 🥳

**Comment by rmichelena** [comment]
This issue gets more relevant now that Matrix is being used for communication with AI Agents.
"hey, check this attached PDF and convert it to Excel" is not possible in a single message. Then if you send the PDF first, the agent does not know what to do with it, and begins analyzing and everything and you have to interrupt it to send the text message. Only way of doing is sending a text message first "hey, I want you to check a PDF and convert it to Excel, stand by, I will send the PDF in the next message" and then wait for the agent to reply "ok, please send the PDF" and then send the PDF - ridiculous, not to mention the waste of time, prompts, and tokens.

**Comment by GenericGoose** [comment]
5  trillion years to implement a basic feature. what do you guys even do lol? no permission system for file uploads either. i guess it's acceptable that everyone gets spammed with gore and csam and that gets federated everywhere else?
